src/store.js

```diff
@@ -0,0 +1,66 @@
+import { makeObservable, observable, action, computed } from "mobx";
+
+export class TodoStoreImplement {
+  todoItems = [];
+  filter = "all";
+
+  constructor() {
+    makeObservable(this, {
+      todoItems: observable,
+      filter: observable,
+      onSubmitTodoContent: action,
+      updateTodoItemStatus: action,
+      handleUpdateStrikeThrough: action,
+      updateFilterStatus: action,
+      todoItemsFiltered: computed,
+    });
+  }
+
+  onSubmitTodoContent(todoInput) {
+    const newItem = {
+      content: todoInput,
+      status: "unchecked",
+      id: this.todoItems.length + todoInput,
+      strikeThrough: false,
+    };
+    this.todoItems.push(newItem);
+  }
+
+  updateTodoItemStatus({ itemId, itemStatus }) {
+    const updatedItemStatus = this.todoItems.map((item) =>
```

Comment:
với mobx thì chỉ nên update đúng vào phần tử cần thay đổi thôi Đức

https://stackoverflow.com/questions/48040627/mobx-mutability-vs-immutability
https://egghead.io/lessons/react-understand-and-control-mutations-with-mobx-dev-tools-and-action-in-react